src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java

```diff
@@ -1389,6 +1464,87 @@ else if (path.toFile().getName().contains("sun")) {
         }
     }
 
+    private static String[] validateRuleNameOrder(String fileName, String lastRuleName,
+                                                  String[] lastRuleNameParts, String ruleName) {
+        final boolean[] beforeFirstSpace = {true};
+        final Long ruleNumberPartsAmountLong = ruleName.chars()
+                .peek(character -> {
+                    if (character == ' ') {
+                        beforeFirstSpace[0] = false;
+                    }
+                })
+                .filter(character -> beforeFirstSpace[0] && character == '.')
+                .count() + 1;
+        final int ruleNumberPartsAmount = ruleNumberPartsAmountLong.intValue();
+        final String[] ruleNameParts = ruleName.split("[. ]", ruleNumberPartsAmount + 1);
```

Comment:
I don't see you doing anything special with the text of the ruleName, you are just getting the number alone.
Why can't you split on the first space  and then with the first string of that, split on the periods and store it into `ruleNamesParts`?

Reply:
I'm not getting the number alone.
The last entry in the array is the text of the rule.
If the number was completely equal, I order by the text of the rule which is in the last array element.
If this is also equal, I error out with "appearing twice", if not, order is defined by the text order.
This resembles the logic present before my PR, just that now the number parts are properly compared numerically and if they were equal, the text is compared as one chunk.

Reply:
> If the number was completely equal, I order by the text of the rule which is in the last array element.

We shouldn't have completely equal numbers. If multiple checks cover 1 section, then both checks will be in that section. We don't drill down into specific sentences of a section.
See http://checkstyle.sourceforge.net/google_style.html#a4.1.2 as an example.

I don't see us needing the text portion so I think it is ok to remove it.

> This resembles the logic present before my PR, just that now the number parts are properly compared numerically and if they were equal, the text is compared as one chunk.

This may have just been lazy coding where we thought we could get by just doing the whole string instead of breaking it out. We obviously didn't go to 10 characters in google's style, so I didn't know 10+ would be an issue.

Reply:
Ok, so I will change it to simply fail after the numbers were equal, not regarding the text portion?

Reply:
If you want to keep the text code for `1.a` etc. that is fine.
I was looking more for making the code listed in the snipped above simplier as I did not understand at all what you were doing until I started debugging it.

Reply:
With text part I meant the text after the space, not textual numberings :-)

Reply:
Oh damn, good that I had to change this, my logic was actually borked for some cases. :-D
Now it should work better and the text part after the first space is not considered anymore.